pkg/operator/vspherecontroller/checks/check_nodes.go

```diff
@@ -169,6 +169,7 @@ func (n *NodeChecker) Check(ctx context.Context, checkOpts CheckArgs) []ClusterC
 
 	// Map of host names and their ESXI versions
 	n.hostESXIVersions = map[string]bool{}
+	n.results = []ClusterCheckResult{}
```

Comment:
The only code change that is needed for the fix. rest of is just tests and stuff.